src/dex/jarvis-v6/jarvis-v6.ts

```diff
@@ -0,0 +1,473 @@
+    const poolIdentifier = eventPool.getIdentifier();
+    if (limitPools && !limitPools.includes(poolIdentifier)) return null;
+
+    const poolState = await this.getPoolState(eventPool, blockNumber);
+
+    const unitVolume = getBigIntPow(
+      (side === SwapSide.SELL ? srcToken : destToken).decimals,
+    );
+
+    const swapFunction = getJarvisSwapFunction(srcToken, eventPool.poolConfig);
+    const maxTokensCapacity = await this.getMaxTokensCapacity(
+      poolAddress,
+      blockNumber,
+    );
+
+    const [unit, ...prices] = this.computePrices(
+      [unitVolume, ...amounts],
+      swapFunction,
+      maxTokensCapacity,
+      side,
+      eventPool.poolConfig,
+      poolState,
+    );
+
+    return [
+      {
+        prices,
+        unit,
+        data: {
+          swapFunction,
+          poolAddress,
+        },
+        poolAddresses: [poolAddress],
+        exchange: this.dexKey,
+        gasCost: 475 * 1000, //between 450-500k gas
+        poolIdentifier,
+      },
+    ];
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { swapFunction } = data;
+    const type = [JarvisSwapFunctions.MINT, JarvisSwapFunctions.REDEEM].indexOf(
+      swapFunction,
+    );
+
+    if (type === undefined) {
+      throw new Error(
+        `Jarvis: Invalid OpType ${swapFunction}, Should be one of ['mint', 'redeem']`,
+      );
+    }
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          opType: 'uint',
+          expiration: 'uint128',
+        },
+      },
+      {
+        opType: type,
+        expiration: (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0),
+      },
+    );
+
+    return {
+      targetExchange: data.poolAddress.toLowerCase(),
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // Encode call data used by simpleSwap like routers
+  // Used for simpleSwap & simpleBuy
+  // Hint: this.buildSimpleParamWithoutWETHConversion
+  // could be useful
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const { swapFunction } = data;
+    const timestamp = (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0);
+
+    let swapFunctionParams: JarvisV6Params;
+    switch (swapFunction) {
+      case JarvisSwapFunctions.MINT:
+        swapFunctionParams = [
+          destAmount,
+          srcAmount,
+          timestamp,
+          this.augustusAddress,
+        ];
+        break;
+      case JarvisSwapFunctions.REDEEM:
+        swapFunctionParams = [
+          srcAmount,
+          destAmount,
```

Comment:
As above, this is a minimum amount, and can be set to 1 to match the contract adapter.